saleor/graphql/order/mutations/orders.py

```diff
@@ -842,6 +850,7 @@ def perform_mutation(cls, _root, info, **data):
         )
 
         recalculate_order(order)
+        update_order_search_document(order)
```

Comment:
Did you consider moving this method call to `Order` model `save` method? This way we wouldn't have to track every single place order gets updated. Currently it is simple and explicit but might turn out to be hard to maintain.

Reply:
Please don't put logic in the `save` method. When `save` is called there is no way to tell why it was called and it's impossible to give it enough context to make use of any data loaders.

Reply:
No, because we do not want to always update `search_document`, but only when field used for searching are changed.